xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -1500,8 +1503,10 @@ void CGUIWindowManager::AddToWindowHistory(int newWindowID)
   { // found window in history
     m_windowHistory = historySave;
   }
-  else
-  { // didn't find window in history - add it to the stack
+  else if (newWindowID != WINDOW_SPLASH)
+  {
+    // didn't find window in history - add it to the stack
+    // but do not add the splash window to history, as we never want to travel back to it
```

Comment:
This is wrong, we we need the splash in history to render it properly. Without `GetActiveWindow` never return the `CGUIWindowSplash` and therefore it's not rendered. We remove the splash window right after the GUI is ready https://github.com/xbmc/xbmc/blob/master/xbmc/Application.cpp#L3928

Reply:
I added this for a reason. If I only could remember what is was. But if you simply revert this another Jack will jump out of the box.

Reply:
If I revert the above line I get "Unable to activate the previous window" in the log if I hit "back" in the startup PVR window. "Back" looks into history, finds WINDOW_SPLASH and tries to go back to the (already removed) WINDOW_SPLASH and then, after the log error, defaults to WINDOW_HOME.

<pre>
(lldb) thread backtrace
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
  * frame #0: 0x00000001010560f0 kodi.bin`CGUIWindowManager::PreviousWindow(this=0x000000010a1068d0) at GUIWindowManager.cpp:709
    frame #1: 0x000000010104584c kodi.bin`CGUIWindow::OnBack(this=0x000000010b0c2e00, actionID=10) at GUIWindow.cpp:964
    frame #2: 0x0000000102628216 kodi.bin`CGUIMediaWindow::OnBack(this=0x000000010b0c2e00, actionID=10) at GUIMediaWindow.cpp:217
    frame #3: 0x0000000101d85502 kodi.bin`PVR::CGUIWindowPVRBase::OnBack(this=0x000000010b0c2e00, actionID=10) at GUIWindowPVRBase.cpp:129
    frame #4: 0x0000000101041226 kodi.bin`CGUIWindow::OnAction(this=0x000000010b0c2e00, action=0x00007ffeefbfd850) at GUIWindow.cpp:449
    frame #5: 0x00000001026277ef kodi.bin`CGUIMediaWindow::OnAction(this=0x000000010b0c2e00, action=0x00007ffeefbfd850) at GUIMediaWindow.cpp:172
    frame #6: 0x0000000101d84d0c kodi.bin`PVR::CGUIWindowPVRBase::OnAction(this=0x000000010b0c2e00, action=0x00007ffeefbfd850) at GUIWindowPVRBase.cpp:113
    frame #7: 0x0000000101d88378 kodi.bin`PVR::CGUIWindowPVRChannelsBase::OnAction(this=0x000000010b0c2e00, action=0x00007ffeefbfd850) at GUIWindowPVRChannels.cpp:132
    frame #8: 0x000000010105a4c8 kodi.bin`CGUIWindowManager::HandleAction(this=0x000000010a1068d0, action=0x00007ffeefbfd850) const at GUIWindowManager.cpp:1094
    frame #9: 0x000000010105a16b kodi.bin`CGUIWindowManager::OnAction(this=0x000000010a1068d0, action=0x00007ffeefbfd850) const at GUIWindowManager.cpp:1044
    frame #10: 0x00000001001bfff0 kodi.bin`CApplication::OnAction(this=0x000000010b800600, action=0x00007ffeefbfd850) at Application.cpp:1998
    frame #11: 0x0000000101186edc kodi.bin`CInputManager::ExecuteInputAction(this=0x000000010a1e5b10, action=0x00007ffeefbfd850) at InputManager.cpp:729
    frame #12: 0x000000010118b099 kodi.bin`CInputManager::HandleKey(this=0x000000010a1e5b10, key=0x00007ffeefbfdca0) at InputManager.cpp:669
    frame #13: 0x0000000101189626 kodi.bin`CInputManager::OnKeyUp(this=0x000000010a1e5b10, key=0x00007ffeefbfe090) at InputManager.cpp:681
    frame #14: 0x00000001011884ad kodi.bin`CInputManager::OnEvent(this=0x000000010a1e5b10, newEvent=0x00007ffeefbfe660) at InputManager.cpp:390
    frame #15: 0x00000001001bf18d kodi.bin`CApplication::HandleWinEvents(this=0x000000010b800600) at Application.cpp:335
    frame #16: 0x00000001001ee31f kodi.bin`CApplication::FrameMove(this=0x000000010b800600, processEvents=true, processGUI=true) at Application.cpp:2670
    frame #17: 0x0000000100485ab6 kodi.bin`CXBApplicationEx::Run(this=0x000000010b800600, params=0x00007ffeefbfeb88) at XBApplicationEx.cpp:108
    frame #18: 0x0000000101a13a61 kodi.bin`::XBMC_Run(renderGUI=true, params=0x00007ffeefbfeb88) at xbmc.cpp:88
</pre>

Reply:
So, we need to remove WINDOW_SPLASH from the history when it gets destroyed (we should do this with every window we destroy, right?!) ... and make sure we are going to WINDOW_HOME, then.

Reply:
Okay will adjust the PR tommorow and remove the destoryed window from history.

Reply:
> remove the destroyed window from history

 Every window we destroy should to be removed from history, not only WINDOW_SPLASH.